test/base/index_range.cpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <memory>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/array.hpp>
+
+
+#include "common/unified/base/kernel_launch.hpp"
+#include "core/base/index_range.hpp"
+#include "core/test/utils.hpp"
+#include "test/utils/executor.hpp"
+
+
+class IndexRange : public CommonTestFixture {
+public:
+    IndexRange() : result_array{exec, 16}, expected_array{ref, 16} {}
+
+    gko::array<int> result_array;
+    gko::array<int> expected_array;
+};
+
+
+// nvcc doesn't like device lambdas declared in complex classes, move it out
+void run_range_for(std::shared_ptr<gko::EXEC_TYPE> exec,
+                   gko::array<int>& result_array)
+{
+    gko::kernels::EXEC_NAMESPACE::run_kernel(
+        exec,
+        [] GKO_KERNEL(auto i, auto result, auto size) {
+            for (auto i : gko::irange<int>{size}) {
+                result[i] = i * i;
+            }
+        },
+        1, result_array, static_cast<int>(result_array.get_size()));
+}
+
+TEST_F(IndexRange, RunsRangeFor)
+{
+    for (auto i : gko::irange<int>{static_cast<int>(result_array.get_size())}) {
```

Comment:
Maybe dont use the irange-based for loop here, and use the regular for loop ? 
```suggestion
    for (int i = 0; i<result_array.get_size; i++) {
```

Reply:
fair point